sdk/jobs/pipelines/1a_pipeline_with_components_from_yaml/pipeline_with_components_from_yaml.ipynb

```diff
@@ -50,7 +49,42 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## 1.2. Configure workspace details and get a handle to the workspace\n",
+    "## 1.2. Configure credential\n",
+    "\n",
+    "We are using `DefaultAzureCredential` to get access to workspace. When an access token is needed, it requests one using multiple identities(`EnvironmentCredential, ManagedIdentityCredential, SharedTokenCacheCredential, VisualStudioCodeCredential, AzureCliCredential, AzurePowerShellCredential`) in turn, stopping when one provides a token.\n",
+    "Reference [here](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.defaultazurecredential?view=azure-python) for more information.\n",
+    "\n",
+    "`DefaultAzureCredential` should be capable of handling most Azure SDK authentication scenarios. \n",
+    "Reference [here](https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity?view=azure-python) for all available credentials if it does not work for you.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from azure.identity import DefaultAzureCredential\n",
+    "\n",
+    "try:\n",
+    "    credential = DefaultAzureCredential(exclude_visual_studio_code_credential=True)\n",
+    "    # Check if given credential can get token successfully.\n",
+    "    credential.get_token('https://management.azure.com/.default')\n",
+    "except Exception as ex:\n",
+    "    # If exception happens when retrieve token, try exclude the failed credential like this then try again:\n",
+    "    # Exclude VSCode credential:\n",
+    "    # credential = DefaultAzureCredential(exclude_visual_studio_code_credential=True)\n",
+    "    # Exclude \n",
+    "    # credential = DefaultAzureCredential(exclude_environment_credential=True)\n",
```

Comment:
when will environment credential be set?

Reply:
It works when some specific environment variables are set, eg AZURE_TENANT_ID, AZURE_CLIENT_ID, etc. ref: https://docs.microsoft.com/en-us/dotnet/api/azure.identity.environmentcredential?view=azure-dotnet